test/core/dns.spec.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const IPFSFactory = require('ipfsd-ctl')
+const IPFS = require('../../src/core')
+
+describe('.dns', () => {
+  let ipfsd, ipfs
+
+  before(function (done) {
+    this.timeout(20 * 1000)
+
+    const factory = IPFSFactory.create({ type: 'proc' })
+
+    factory.spawn({
+      exec: IPFS,
+      initOptions: { bits: 512 },
+      config: { Bootstrap: [] }
+    }, (err, _ipfsd) => {
+      expect(err).to.not.exist()
+      ipfsd = _ipfsd
+      ipfs = _ipfsd.api
+      done()
+    })
+  })
+
+  after((done) => {
+    if (ipfsd) {
+      ipfsd.stop(done)
+    } else {
+      done()
+    }
+  })
+
+  // skipping because there is an error in https://ipfs.io/api/v0/dns?arg=ipfs.io
+  // unskip once this is resolved: https://github.com/ipfs/go-ipfs/issues/6086
```

Comment:
this is resolved in https://github.com/ipfs/go-ipfs/pull/6087

Reply:
Hi @jchris, 

When I go to https://ipfs.io/api/v0/dns?arg=ipfs.io   or https://ipfs.io/api/v0/dns?arg=ipfs.io&recursive=false (IPFS companion turned off) It still gives me an error. 

Are the changes from # 6087 integrated into the ipfs.io api?

Reply:
I don't know, I just know it was closed and merged. Trying to keep the links current.

Reply:
Okay. Thank you. :)

Reply:
It was fairly recent - I think it'll be released in 0.4.20